src/core/org/apache/jmeter/gui/UndoHistory.java

```diff
@@ -118,8 +87,12 @@ public void clear() {
             return;
         }
         log.debug("Clearing undo history");
-        history.clear();
-        position = INITIAL_POS;
+        manager.discardAllEdits();
+        if (isTransaction()) {
+            log.warn("Clearing undo history with " + transactions.size() + " unfinished transactions");
```

Comment:
I like the logs to use the formatting/placeholder feature `{}` together with parameters.